### DISCUSSION TITLE
laboratory management system

### DISCUSSION THREAD

**Discussion by nsrmhjl** [nsrmhjl/ptDtSc#11] [Q&A]
سلام @AlirezaDoustmohammadi 
خوبی خوشی؟
میگم به نظرت این خوبه:
[labm](https://github.com/sohailmahmud/labm
)

**Comment by nsrmhjl** [comment]
فکر کنم این هم ویدیو مربوط به همین برنامه باشه:
[https://www.youtube.com/watch?v=Cg4UFW9QGLo](https://www.youtube.com/watch?v=Cg4UFW9QGLo)

**Comment by nsrmhjl** [comment]
به نظرت ران کردن این چقدر پام آب می خوره؟
[https://github.com/elabftw/elabftw](https://github.com/elabftw/elabftw)

به نظر میاد بهترین برنامه تو کتگوری خودش هست.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
به نظرم دکتر بستگی به این داره که چه ویژگی ای براتون بیشتر اهمیت داره.

تو برنامه labm مدیریت منابع به نظرم اومد بهتره. مثلا مشخصه از فلان وسیله آزمایشگاهی در حال حاضر چند تا موجوده و احتمالا وقتی یکی استفاده میکنه و تو سیستم ثبت میشه، خودش میاد quantity رو کم میکنه.

اینو من تو برنامه elabfw ندیدم. 
ولی تو این برنامه مدیریت پروژه خیلی خوبه و هر تسکی که انجام میشه رو خیلی خوب میشه مدیریت کرد. یه بخشی هم به نام   دیتابیس داشت که نفهمیدم منظورش چیه. مثال هایی که تو دموش تو این بخش قرار داده بود چیز خاصی نداشت.

هزینه ران کردن  elabfw ماهی حدود 200 هزار تومن هست

**Reply by nsrmhjl** [reply]
سلام ای دوست 
یعنی هزینه اجاره سرور و نصب داکر و ران کردن داکر ایمیج elabfw ماهی ۲۰۰ تومن میشه؟

**Comment by AlirezaDostmohammadi** [comment]
نکته مهم دیگه هم که وجود داره اینکه  elabfw خیلی پشتیبانی بهتری داره و زود به زود براش آپدیت میدن

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر. تنها بخشی که هزینه داره اجاره سروره.
که ماهی ۲۰۰ تومن میشه

**Reply by nsrmhjl** [reply]
خوب اینکه خیلی خوبه. 
من چیکار باید بکنم؟
می تونم از همین دومین خودم به این سرور اجاره ای راه بدم؟ مثلا lab.mohajelresearch.net

**Reply by nsrmhjl** [reply]
آیا هم باید پول کرایه هاست بدم و هم پول کرایه سرور؟

**Reply by nsrmhjl** [reply]
@AlirezaDoustmohammadi 
سلام
دیدی پیاممو.

**Reply by AlirezaDostmohammadi** [reply]
> آیا هم باید پول کرایه هاست بدم و هم پول کرایه سرور؟

بله دکتر. البته میشه جفتش رو روی سرور مجازی چید که پول یکی شو بدید. بعد رانه برنامه مشخص میشه که سرور کشش اش رو داره که روش سایت هم بالا بیاد یا نه

**Comment by AlirezaDostmohammadi** [comment]
<div dir="rtl">

سلام دکتر خوبید؟
بله دیدم و جواب دادم. بعد یه روز جیمیل زد  draft شده و نرفته!
دکتر اینو پیگیری میکنم. فقط سرور از ایران مد نظرتونه یا خارج از ایران؟
ایران حدود 200 در میاد. خارج ایران تا حدود 300 تا 400

**Comment by nsrmhjl** [comment]
ممنون از شما
نه تو همین خراب شده هم کافیه

**Comment by nsrmhjl** [comment]
تو داکیومنتیشن گفته مقدار رم و استوریج مورد نیازش رو. بک گیگ رم مثلا براش
خوبه. اگر کمی بیشتر از اون بگیرم فکر کنم بتونم سایت رو هم ران کنم. ممنونم
که پیگیری می کنی.
الان خیلی وضع خرابی دارم. نمیدونم چه ماده ای رو کجا دارم. دانشجو ها نوشتن
گزارش رو می پیچونن و هر چیزی رو باید از خودشون دوباره بپرسم.

**Comment by AlirezaDostmohammadi** [comment]
بله دکتر مقدار استفاده از رمش 1 گیگ هست ولی چون روی داکر اجرا میشه، باید میزان رم داکر رو هم در نظر بگیریم که نوشته حداقل 4 گیگ. 
از شرکت پرسیدم جواب دادن، اطلاع میدم

**Comment by AlirezaDostmohammadi** [comment]
سلام آقای دکتر. خوبید؟
آقای دکتر الان هم سایتتون بدون فیلتر شکن بالا نمیاد؟
من تست کردم، برای من بالا میاد بدون فیلتر شکن

**Reply by nsrmhjl** [reply]
سلام
نه تو پاستور بالا نمیاد
با اینترنت موبایل هم هات اسپات کردم و بالا نمیاد.

**Reply by nsrmhjl** [reply]
با فیلتر شکن راحت باز میشه. نمی دونم شاید مشکل از کامپیوتر خودمه. من dns م ۱.۱.۱.۱ هستش

**Comment by AlirezaDostmohammadi** [comment]
تو گوشی تون هم همینطوره دکتر؟

**Reply by nsrmhjl** [reply]
تو گوشیم باز میشه!

**Comment by AlirezaDostmohammadi** [comment]
احتمالا پس دکتر مشکل از  dns باشه.

**Reply by nsrmhjl** [reply]
آره. تو dns گوگل داری؟ یا اصلا اون قسمت رو خالی گذاشتی؟

**Comment by AlirezaDostmohammadi** [comment]
خالی میزارم دکتر

**Comment by nsrmhjl** [comment]
سلام دوست محمدی
من پاسخ اونها به تیکت هام رو برات فوروارد کردم.
دیدیشون؟ گفته می تونیم بیاریمش ایران. 
به نظرت چیکار کنم؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان خوبید؟
به نظرم بگیریم سرور مجازی از ایران رو. من انجام بدم یا خودتون زحمتش رو میکشید؟

**Comment by nsrmhjl** [comment]
ممنون میشم زحمتشو بکشی

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. کارش انجام شد.
فقط تحویل ندادن.
نوشته:
سپاس از خرید شما : تحویل اولین خرید سرور مجازی ایران نیاز به احراز هویت دارد در بازه تحویل اعلام شده در این خصوص با شما تماس حاصل شده و پس از احراز هویت، مشخصات سرویس از طریق ایمیل برای شما ارسال می شود. 
لطفا زمان تماس مجدد بهشون تاکید کنید که اوبونتو  desktop روی سرور نصب بشه.

**Reply by nsrmhjl** [reply]
سلام 
خیلی ممنون
چشم حتما
شما مبلغ به علاوه شماره کارتت رو همینجا بنویس بی زحمت.

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان.
فاکتور پیوست شد.
[پارس وی دی اس _ PARSVDS.pdf](https://github.com/nsrmhjl/ptDtSc/files/10301866/_.PARSVDS.pdf)

شماره کارت:
6037691666716006

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
خدمت شما

https://lab.mohajelresearch.net

**Comment by AlirezaDostmohammadi** [comment]
حتما باید ابتداش  https بزنید و الا بالا نمیاد.

**Comment by nsrmhjl** [comment]
سلام 
ایول خیلی باحاله 
باید خودمم رجیستر کنم توش؟ ادمین نداره؟

**Comment by AlirezaDostmohammadi** [comment]
تا جایی که من میدونم نه دکتر جان نداره. اولین کسی که در گروه عضو میشه، میشه
ادمین

**Comment by nsrmhjl** [comment]
دمت گرم. خیلی عالی شد. باید نگهداریش رو ازت یاد بگیرم سر فرصت.
بک آپ
آپدیت
نحوه تمدید و ...

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان

**Reply by nsrmhjl** [reply]
سلام ای دوست @AlirezaDoustmohammadi 
خوبی؟
رسیدی بک آپ و آپدیت و ... این برنامه رو بخونی؟

**Comment by nsrmhjl** [comment]
سلام ای دوست خوبی؟
میگم میشه بهم یاد بدی که در این موارد:
[setting-up-email](https://doc.elabftw.net/sysadmin-guide.html#setting-up-email)
[how-to-backup](https://doc.elabftw.net/backup.html#how-to-backup)

و همینطور آپدیت کردن چیکار باید بکنم؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جلن. خوبید؟

حتما چشم. یه چک کنم، خدمتتون میگم

**Comment by nsrmhjl** [comment]
اینجا من سوال و چیزی ندیدم
میشه دوباره پست بذاری؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. حالتون چطوره خوبید؟

**Comment by AlirezaDostmohammadi** [comment]
دکتر بچه های آزمایشگاه اطلاعاتی روی سایت وارد کردن که من بتونم کامل بک آپ و .. چک کنم؟
لطفا یه اکانت در اختیار من قرار میدید با دسترسی کامل برای اینکه سایت رو چک کنم کامل

**Reply by nsrmhjl** [reply]
سلام 
با ایمیل خودت ساختم
پسورد اس شد
اما توش خبری نیست
نه آپدیت میشه و نه بک میگیره
همه چیش از ترمینال هست.

**Reply by nsrmhjl** [reply]
بله یک چیزهایی وارد کردم
بچه ها شامل خودم میشن.

**Comment by AlirezaDostmohammadi** [comment]
😅 خیلی ممنون دکتر جان

**Comment by nsrmhjl** [comment]
سلام ای دوست 
@AlirezaDoustmohammadi 

من خیلی گیر این هستم. اگر بهم یادش بدی دیگه خودم از اون به بعدش رو انجام میدم. یعنی مراحل ایمیل، بک آپ و به روز رسانیش رو

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟

حتما روش کار میکنم. فقط نمیدونم چرا پسوردی که فرستادید برام کار نمیکنه.

کلا یه بار دیگه باید از صفر config اش کنم.
چون یکم حالت وصله پینه شده، نگهداری اش سخته به نظرم.

یه نکته ای هم که هست دکتر اون داستان ایمیل فکر میکنم تو ایران تحریمه.
مجدد چکش میکنم

**Comment by nsrmhjl** [comment]
ممنونم

یه پس جدید فرستادم 

میگم الان دیگه اطلاعات زیادی داریم توش

میترسم اگه از اول کانفیگ کنی بپره

**Comment by AlirezaDostmohammadi** [comment]
خیلی ممنون دکتر. پس یه راه دیگه ای پیدا میکنم که اطلاعات نپره

**Comment by nsrmhjl** [comment]
من می تونم دونه دونه آیتم ها رو ازش اکسپورت کنم 

اگر راهی نداره بهم بگو تا همین کارو بکنم.

**Comment by nsrmhjl** [comment]
البته یک راه بک آپ آسون از دیتابیسش از ترمینال هم داره.
می خوای امتحانش کنم؟

**Comment by AlirezaDostmohammadi** [comment]
میتونییم تستش کنیم دکتر.
چون به هر حال باید بک آپ بگیریم

**Comment by nsrmhjl** [comment]
نه مثل اینکه اشتباه می کردم 
اینقدر ها هم آسون نیست
باید اول borg رو نصب کنم. بعد ست آپش کنیم بک آپ رو. همش از طریق ترمینال به سرور هست. حالا میشه آدرس خودش رو بدیم (یعنی توی خود سرور بک آپ بگیریم) یا یک سرور دیگه رو.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر. بخش backupدرست شد.
الان با دستور 
elabctl backup
بک آپ گرفته میشه. البته من تنظیمش کردم هر روز ساعت 4 صبح بک آپ بگیره. ولی باید صبر کنیم ببینیم درست کار میکنه یا نه.

نحوه restore کردن بک آپ ها رو اینجا نوشته:

https://doc.elabftw.net/backup.html#how-to-restore-a-backup

**Comment by AlirezaDostmohammadi** [comment]
نحوه آپدیت:
1. اول بهتره backupگرفته بشه.
2. از طریق گیتهاب https://github.com/elabftw/elabftw باید ببینیم آخرین ورژن برنامه چیه. مثلا الان آخرین ورژن، ورژن 4.5.14 هست.
3. داخل سرور در دایرکتوری 
/etc
یک فایلی وجود داره به نام:
elabftw.yml
بازش میکنیم.
تو خط 16 نوشته شده:
image: elabftw/elabimg:X.Y.Z
به جای X.Y.Z ورژن برنامه رو مینویسیم. مثلا:
image: elabftw/elabimg:4.5.14

بعد اعمال این تغییر در فایل elabftw.yml، کامند های زیر رو اجرا میکنیم:
  elabctl update
که باعث میشه آخرین image از داکر دانلود بشه.
 elabctl self-update
که اسکریپت های elabctl رو آپدیت میکنه.
و در انتها:
docker exec -it elabftw bin/console db:update

**Reply by AlirezaDostmohammadi** [reply]
https://doc.elabftw.net/how-to-update.html

**Comment by AlirezaDostmohammadi** [comment]
دکتر بخش ایمیل هم اوکی شد.
فکر کنم براتون یه ایمیل اومد.
روی دکمه forgot password هم زدم و اوکی بود.

نحوه ست کردن ایمیل:
باید به پنل sysadmin در پروفایل خودتون برید.


![image](https://user-images.githubusercontent.com/49092099/230595002-2e5e5653-921b-478d-8011-888969120303.png)

و در صفحه ای که باز میشه، روی تب عه Email کلیک کنید:

![image](https://user-images.githubusercontent.com/49092099/230595064-593086e4-4797-4542-8b24-ceca1f340464.png)

و در صفحه جدیدی که باز میشه، اطلاعات زیر رو وارد کنید:
![image](https://user-images.githubusercontent.com/49092099/230595148-226e9af1-9a27-4d27-9cb0-90868ade76d6.png)


تو قسمت SMTP Password هم این پسورد رو باید بزنید:
yt^uoZ-ELRoS

من ایمیل رو از طریق اکامن خودم که برام ایجاد کرده بودید ساختم. فعلا نیاز نیست این مراحل انجام بدید اما بعدا هر وقت احتیاج شد، میتونید از طریق پروفایل خودتون تنظیمات رو انجام بدید.

بعد

**Comment by AlirezaDostmohammadi** [comment]
به نظرم الان دیگه کامل اوکی هست سایت دکتر.
فقط داستان SSL میمونه که تو گیتهابشون میپرسم.

**Comment by nsrmhjl** [comment]
ایول دمت گرم. خیلی لطف کردی.
یبار که اومدی مراحلش رو با هم میریم و یا یه فیلم مثل اون مودل برام درست بکنی دیگه توپ توپ میشم.

**Reply by AlirezaDostmohammadi** [reply]
قربانت دکتر جان. حتما

**Comment by nsrmhjl** [comment]
راستی بک آپ قبلی رو اور رایت می کنه یا در کنارش بک آپ میگیره؟

**Reply by AlirezaDostmohammadi** [reply]
دکتر فردا اینو میتونیم مطمئن بشیم. چیزی که مشخصه اینکه بک آپ هایی که در یک روز گرفته میشه، overwrite میشن. حالا اینکه اگه در روزهای مختلف بگیریم، بازهم overwrite میشن یا نه رو فردا متوجه میشیم.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر.
مشکل unsecure بودن کانکشن به سایت حل شد.
بک آپ هم اگر به صورت در روزهای مختلف گرفته بشه، overwriteنمیشه.

**Reply by nsrmhjl** [reply]
ببخشید که دیر جواب میدم.
خیلی ممنون از لطفت.
یک سری سوال راجع به بک آپ هست که هر وقت اومدی ازت می پرسم.
بازم ازت ممنونم.

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان. حتما

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. خوبید؟

دکتر parsvds یه تخفیف 50 درصد برای تمدید هاست هاش گذاشته. خواستم اطلاع بدم که اگر قصد تمدید دارید، میتونید بهشون تیکت بزنید که فاکتور براتون صادر کنن

**Comment by nsrmhjl** [comment]
سلام ای دوست
چطوری؟
فقط هاست یا سرور هم جزوش هست؟ من الان ماهی سیصد میدم به سرور.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. قربانت. شما خوبی؟

فک کنم دکتر هاست هم شامل میشه. فقط دامنه شامل نمیشه.
بهشون تیکت بزنید، براتون فاکتور صادر میکنن

**Comment by nsrmhjl** [comment]
یه سوال
آیا می تونم مودل رو هم بیارم روی همین سرور مجازی؟

**Comment by AlirezaDostmohammadi** [comment]
دکتر احتمالا میشه. باید تست کنیم این برنامه ای که رو سرور هست downنشه.
چون برای بالا آوردن مودل باید آپاچی نصب بشه و تنظیمات سرور رو بزنیم به هم.

**Comment by nsrmhjl** [comment]
سلام @AlirezaDoustmohammadi 
شرمنده هی زحمت میدم.
میشه برای 
backup
restore
update
این سرور برای من یک فیلم آموزشی درست کنی که من از روش تمرین کنم. 
تو که بری اگر خودم این کار ها رو انجام ندم سیستم از هم می پاشه.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر عزیز خوبید؟
اختیار دارید دکتر جان. حتما

**Reply by nsrmhjl** [reply]
بسیار ممنون

**Comment by AlirezaDostmohammadi** [comment]
نحوه آپدیت کردن elabftw:

https://mohajelresearch.net/elab/update-elabftw.mp4

دکتر جان ویدیوهارو تو هاست خودتون اپلود کردم که دسترسی اش براتون همیشگی باشه.

سایت هایی که تو ویدیو درباره اش صحبت شده:
https://shecan.ir/
https://doc.elabftw.net/how-to-update.html
https://github.com/elabftw/elabctl
https://github.com/elabftw/elabftw

فایل هایی که تو ویدیو درباره اش صحبت شد:
[elabftw.yml.zip](https://github.com/nsrmhjl/ptDtSc/files/12321952/elabftw.yml.zip)
[elabctl.zip](https://github.com/nsrmhjl/ptDtSc/files/12321971/elabctl.zip)

**Reply by AlirezaDostmohammadi** [reply]
@nsrmhjl نحوه update  کردن

**Comment by AlirezaDostmohammadi** [comment]
نحوه backup گرفتن از elabftw:
https://mohajelresearch.net/elab/backup-elabftw.mp4

سایت هایی که تو ویدیو درباره اش صحبت شده:
https://doc.elabftw.net/backup.html#with-elabctl
https://github.com/elabftw/elabctl/tree/master

فایلی که تو ویدیو درباره اش صحبت شده:
[elabctl.conf.zip](https://github.com/nsrmhjl/ptDtSc/files/12321819/elabctl.conf.zip)

**Comment by AlirezaDostmohammadi** [comment]
نحوه restore کردن backup های گرفته شده:
https://mohajelresearch.net/elab/restore-elabftw.mp4


داکیومنتی که تو ویدیو بهش اشاره کردم:
https://doc.elabftw.net/backup.html#how-to-restore-a-backup

**Comment by nsrmhjl** [comment]
سلام
خیلی ازت ممنونم. دوشنبه راجع به این هم صحبت کنیم. 
آپدیت رو هم در دست تهیه داری؟

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان. حتما. نه دکتر دیروز فرستادم اون رو هم
